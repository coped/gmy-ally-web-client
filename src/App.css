.App {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: #eeeeee;
    background-color: #2d4059;
}

.logo-medium {
    width: 50vh;
}