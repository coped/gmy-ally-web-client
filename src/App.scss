#App {
    height: 100vh;

    .welcome-title {
        display: flex;
        justify-content: space-around;
        font-weight: light;
        font-size: 3rem;

    }
}