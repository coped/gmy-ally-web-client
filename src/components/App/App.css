.App {
  text-align: center;
}

.App-logo {
  width: 60vh;
  height: auto;
  pointer-events: none;
}

p {
  margin: 1rem auto;
}


@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-breathe alternate infinite 1.5s;
  }
}


.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #eeeeee;
  background-color: #2d4059;
}

@keyframes App-logo-breathe {
  from {
    transform: scale(1, 1);
  }
  to {
    transform: scale(1.2, 1.2);
  }
}

