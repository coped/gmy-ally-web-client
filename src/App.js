import React, { useState } from "react";
import "./App.scss";
import "bulma/css/bulma.css";
import { Login, Signup, Welcome, Dashboard } from "pages";
import { AuthContext, useAuth } from "context/auth";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
  useHistory,
  useLocation,
} from "react-router-dom";
import PrivateRoute from "PrivateRoute";

export default function App(props) {
  const existingTokens = JSON.parse(localStorage.getItem("tokens"));
  const [authTokens, setAuthTokens] = useState(existingTokens);

  function setTokens(data) {
    localStorage.setItem("tokens", JSON.stringify(data));
    setAuthTokens(data);
  }

  return (
    <AuthContext.Provider
      value={{ authTokens: authTokens, setAuthTokens: setTokens }}
    >
      <div id="App" className="App container">
        <Router>
          <div className="welcome-title">
            <h1>
              Welcome to <a href="/">Gym Partner</a>
            </h1>
          </div>
          <Route exact component={Welcome} path="/" />
          <Route component={Login} path="/login" />
          <Route component={Signup} path="/signup" />
          <PrivateRoute component={Dashboard} path="/dashboard" />
        </Router>
      </div>
    </AuthContext.Provider>
  );
}
