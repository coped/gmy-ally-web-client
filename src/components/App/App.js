import React, { Component } from 'react';
import gymPartnerLogo from 'assets/images/logo_transparent.png';
import './App.css';

class App extends Component {
  componentDidMount() {
    fetch("/api/v1/users.json")
    .then(response => response.json())
    .then(myJson => console.log(myJson))
    .catch(error => console.error('Error:', error));
  }

  render () {
    return (
      <div className="App">
        <div className="App-header">
          <img src={gymPartnerLogo} className="App-logo" alt="logo" />
          <p>
            Gym Partner is currently under construction.
          </p>
          <p>
            Visit our <a href="https://github.com/coped/gym-partner-web-client">
              <u>Github</u>
            </a> for updates.
          </p>
        </div>
      </div>
    );
  }
}

export default App;
