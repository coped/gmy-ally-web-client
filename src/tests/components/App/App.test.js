import React from "react";
import { render } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App, { Login, Dashboard } from "App";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Login component", () => {
    const wrapper = mount(<App />);
    console.log(wrapper.debug());
    expect(wrapper.find(Login).length).toEqual(1);
  });
});
