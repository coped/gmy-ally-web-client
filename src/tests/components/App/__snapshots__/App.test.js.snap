// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App container"
  id="App"
>
  <div
    className="welcome-title"
  >
    <h1>
      Welcome to 
      <a
        href="/"
      >
        Gym Partner
      </a>
    </h1>
  </div>
  <div
    id="Welcome"
  >
    <a
      href="/signup"
    >
      <button
        className="button is-primary"
      >
        Sign up
      </button>
    </a>
    <a
      href="/login"
    >
      <button
        className="button is-link"
      >
        Log in
      </button>
    </a>
  </div>
</div>
`;
